src/target/linux.rs

```diff
@@ -56,6 +67,47 @@ pub struct AttachOptions {
     pub kill_on_exit: bool,
 }
 
+pub struct Watchpoint {
+    typ: WatchpointType,
+    addr: usize,
+    size: WatchSize,
+    on_trap: std::boxed::Box<dyn FnMut(&Self)>,
```

Comment:
Treating this like a software breakpoint, aka returning a signal from `step`/`unpause` is more flexible, as the user of headcrab can for example allow the enduser to inspect/modify the program state before continuing.

Reply:
Ok sure, will remove `on_trap`